BoxContentSDK/BoxContentSDK/Models/BOXRepresentation.m

```diff
@@ -0,0 +1,77 @@
+//
+//  BOXRepresentation.m
+//  BoxContentSDK
+//
+//  Created on 7/15/16.
+//  Copyright © 2016 Box. All rights reserved.
+//
+
+#import "BOXRepresentation.h"
+#import "NSJSONSerialization+BOXAdditions.h"
+#import "BOXContentSDKConstants.h"
+
+@implementation BOXRepresentation
+
+- (instancetype)initWithJSON:(NSDictionary *)JSONResponse
+{
+    if (self = [super init]) {
+        
+        self.JSONData = JSONResponse;
+        
+        self.type = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyRepresentation
+                                                   inDictionary:JSONResponse
+                                                hasExpectedType:[NSString class]
+                                                    nullAllowed:NO];
+        
+        NSDictionary *propertiesJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyProperties
+                                                                  inDictionary:JSONResponse
+                                                               hasExpectedType:[NSDictionary class]
+                                                                   nullAllowed:NO];
+        
+        self.dimensions = [propertiesJSON objectForKey:BOXAPIObjectKeyDimensions];
+        
+        self.status = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyStatus
+                                                     inDictionary:JSONResponse
+                                                  hasExpectedType:[NSString class]
+                                                      nullAllowed:NO];
+        
+        NSDictionary *detailsJSON = [NSJSONSerialization box_ensureObjectForKey:BOXAPIObjectKeyDetails
+                                                                   inDictionary:JSONResponse
+                                                                hasExpectedType:[NSDictionary class]
+                                                                    nullAllowed:NO];
+        if ([detailsJSON count] > 0) {
```

Comment:
If there are no details, the API still returns an empty dictionary, so thought we should check to see if it has anything in it before setting the property. Is it better for 1) the API to return null if there's nothing in it or 2) just set it as an empty dictionary?